app/src/main/report/sources/ReportingSource.ts

```diff
@@ -0,0 +1,34 @@
+import {FetchConfig, Source, UrlBuilder} from "../../shared/sources/Source";
+import {makeNotificationException, notificationActions} from "../../shared/actions/NotificationActions";
+import {ReportingFetcher} from "./ReportingFetcher";
+import fetcher from "../../shared/sources/Fetcher";
+
+export abstract class ReportingSource<TState> extends Source<TState> {
+
+    protected doFetch<TModel>(urlFragment: UrlBuilder<TState>, config: FetchConfig<TState, TModel>): AltJS.SourceModel<TModel> {
+        const handler = this.processResponse;
+        const source: AltJS.SourceModel<TModel> = {
+            remote(state: TState) {
+                return handler((<ReportingFetcher>fetcher.fetcher).fetchFromReportingApi(urlFragment(state)))
```

Comment:
the only difference between this and the Source base class is this line, that casts the fetcher to a ReportingFetcher, and calls through to the Orderly API instead of the Montagu API.